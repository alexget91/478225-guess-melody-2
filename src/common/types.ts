import {QuestionType} from "./constants";

type Song = {
  artist: string,
  picture: string,
};

type ArtistAnswer = {
  artist: string,
  picture: string,
};

type GenreAnswer = {
  genre: string,
  src: string,
};

type ArtistQuestion = {
  type: QuestionType,
  song: Song,
  answers: Array<ArtistAnswer>,
};

type GenreQuestion = {
  type: QuestionType,
  genre: string,
  answers: Array<GenreAnswer>,
};

type User = {
  id: number,
  email: string,
};

export {Song, ArtistQuestion, GenreQuestion, User};
