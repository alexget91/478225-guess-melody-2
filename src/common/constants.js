const ScreenSteps = {
  START: -1,
  FAIL_TIME: -2,
  FAIL_MISTAKES: -3,
};

export {ScreenSteps};
