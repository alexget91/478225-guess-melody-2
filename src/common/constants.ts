enum ScreenStep {
  START = -1,
}

enum QuestionType {
  ARTIST = `artist`,
  GENRE = `genre`,
}

enum FailType {
  TIME = `TIME`,
  MISTAKES = `MISTAKES`,
}

export {ScreenStep, QuestionType, FailType};
