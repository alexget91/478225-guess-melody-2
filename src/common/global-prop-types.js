import PropTypes from "prop-types";

const genreQuestion = {
  type: PropTypes.oneOf([`genre`, `artist`]).isRequired,
  genre: PropTypes.string.isRequired,
  answers: PropTypes.arrayOf(PropTypes.exact({
    genre: PropTypes.string.isRequired,
    src: PropTypes.string.isRequired,
  })).isRequired,
};

const artistQuestion = {
  type: PropTypes.oneOf([`genre`, `artist`]).isRequired,
  song: PropTypes.exact({
    artist: PropTypes.string.isRequired,
    src: PropTypes.string.isRequired,
  }).isRequired,
  answers: PropTypes.arrayOf(PropTypes.exact({
    artist: PropTypes.string.isRequired,
    picture: PropTypes.string.isRequired,
  })).isRequired
};

const userData = {
  id: PropTypes.number.isRequired,
  email: PropTypes.string.isRequired,
};

export {genreQuestion, artistQuestion, userData};
