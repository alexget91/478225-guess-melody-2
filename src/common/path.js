const Path = {
  INDEX: `/`,
  AUTH: `/auth`,
  WIN: `/win`,
  LOSE_MISTAKES: `/lose-mistakes`,
  LOSE_TIME: `/lose-time`,
};

export default Path;
