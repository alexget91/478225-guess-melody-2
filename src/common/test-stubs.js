const createAudioMock = () => {
  return {createNodeMock(element) {
    if (element.type === `audio`) {
      return {
        createRef() {}
      };
    }
    return null;
  }};
};

export default createAudioMock;
