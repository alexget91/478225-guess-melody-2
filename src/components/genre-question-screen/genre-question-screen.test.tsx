import * as React from "react";
import * as renderer from "react-test-renderer";
import GenreQuestionScreen from "./genre-question-screen";
import {QuestionType} from "../../common/constants";

it(`Genre question screen correctly renders after relaunch`, () => {
  const tree = renderer
    .create(<GenreQuestionScreen
      screenIndex={0}
      question={{
        type: QuestionType.GENRE,
        genre: ``,
        answers: [{
          genre: ``,
          src: ``,
        }],
      }}
      userAnswer={[]}
      renderPlayer={jest.fn()}
      onAnswerChange={jest.fn()}
      onAnswerSubmit={jest.fn()}
    />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
