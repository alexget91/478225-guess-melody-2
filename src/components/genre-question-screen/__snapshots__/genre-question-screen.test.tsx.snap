// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Genre question screen correctly renders after relaunch 1`] = `
<section
  className="game__screen"
>
  <h2
    className="game__title"
  >
    Выберите 
    
     треки
  </h2>
  <form
    className="game__tracks js-answer-form"
    onSubmit={[Function]}
  >
    <div
      className="track"
    >
      <div
        className="game__answer"
      >
        <input
          className="game__input visually-hidden"
          data-index={0}
          id="answer-0"
          name="answer"
          onChange={[MockFunction]}
          type="checkbox"
          value="answer-0"
        />
        <label
          className="game__check"
          htmlFor="answer-0"
        >
          Отметить
        </label>
      </div>
    </div>
    <button
      className="game__submit button"
      type="submit"
    >
      Ответить
    </button>
  </form>
</section>
`;
